### PR TITLE
Require passphrase to confirm account deletes or resets

### PR DESCRIPTION
**Pull request by joshblum**

### CODE REVIEW

**Comment by maxtaco** [comment]
Does the server enforce this too?

File: go/libkb/login_state.go [hunk 1]
```diff
@@ -461,10 +461,7 @@ func (s *LoginState) resetOrDelete(un string, which string) (err error) {
 			SessionR:    lctx.LocalSession(),
 		}
 		pdpka.PopulateArgs(&arg.Args)
-		res, aerr := s.G().API.Post(arg)
-		if aerr == nil {
-			s.G().Log.Info("%s Result: %+v\n", which, res.AppStatus)
```
Comment (joshblum) [inline go/libkb/login_state.go]: removed so it doesn't print to stdout we can `Log.Debug` if anyone feels strongly

**Comment by patrickxb** [comment]
A few comments:

1. I don't think using LoginWithPrompt is the right call as the GUI won't be expecting the LoginUI calls at the point where people press "Delete account" in the GUI.  I thought the account delete engine could call SecretUI.GetPassphrase() then use LoginState.VerifyPlaintextPassphrase().

2. The reason to have delete account all in one LoginState request is so that it is serial.  The way it is in your PR is in three pieces, so something could come in between those and screw things up, so I think all the guts of this should stay in one LoginState request.

Happy to discuss in person if easier.

**Comment by joshblum** [comment]
Makes sense, let me know if the refactor works and otherwise we can discuss in person

In the ticket we mentioned allowing this command outside of devel -- should we remove this from 
https://github.com/keybase/client/blob/master/go/client/commands_devel.go#L18? Are there any other steps or do we want to do that at a later time?

**Comment by patrickxb** [comment]
Looks good to me.  Have you tried it in GUI and it works?

**Comment by joshblum** [comment]
Added confirmation for acctreset cmd and verified GUI requires confirmation too

**Pull request merged**